app/api/monthly.py

```diff
@@ -0,0 +1,448 @@
+        "type": "weekly"
+    })
+    ret = [serialize_doc(doc) for doc in ret]
+    return ret
+
+
+def load_user(user):
+    ret = mongo.db.users.find_one({
+        "_id": ObjectId(user)
+    }, {"profile": 0})
+    return serialize_doc(ret)
+
+# function to add user data
+def add_user_data(user):
+    user_data = user['user']
+    user_data = (load_user(user_data))
+    user['user'] = user_data
+    return user
+
+# function for loading manager data
+def load_manager(manager):
+    ret = mongo.db.users.find_one({
+        "_id": manager
+    }, {"profile": 0})
+    return serialize_doc(ret)
+
+# function for adding manager data
+def add_manager_data(manager):
+    for elem in manager['review']:
+        elem['manager_id'] = load_manager(ObjectId(elem['manager_id']))
+    return manager
+
+# function to load details of user
+def load_details(data):
+    user_data = data['user']
+    all_weekly = data['user']
+    all_weekly = (load_all_weekly(all_weekly))
+    user_data = (load_user(user_data))
+    data['user'] = user_data
+    if 'review' in data:
+        review_detail = data['review']
+    else:
+        review_detail = None
+    if review_detail is not None:
+        for elem in review_detail:
+            elem['manager_id'] = load_manager(ObjectId(elem['manager_id']))
+    data['all_weekly'] = all_weekly
+    return data
+
+# function where no review will be shown to manager
+def no_review(data):
+    user_data = data['user']
+    user_data = (load_user(user_data))
+    data['user'] = user_data
+    review_data = None
+    data['review'] = review_data
+    return data
+
+
+# Api for delete monthly report
+@bp.route('/delete_monthly/<string:monthly_id>', methods=['DELETE'])
+@jwt_required
+def delete_monthly(monthly_id):
+    current_user = get_current_user()
+    docs = mongo.db.reports.remove({
+        "_id": ObjectId(monthly_id),
+        "type": "monthly",
+        "user": str(current_user['_id'])
+    })
+    return jsonify(str(docs)), 200
+
+
+# Api for monthly checkin
+@bp.route('/monthly', methods=["POST", "GET"])
+@jwt_required
+def add_monthly_checkin():
+    today = datetime.datetime.utcnow()
+    month = today.strftime("%B")
+    current_user = get_current_user()
+    doj = str(current_user['dateofjoining'])
+    slack = current_user['slack_id']
+    # convert join data in string strip time date and fetch just the date variable from whole date
+    date = datetime.datetime.strptime(doj, "%Y-%m-%d %H:%M:%S")
+    datee = date.day
+    if request.method == "GET":
+        report = mongo.db.reports.find({
+            "user": str(current_user["_id"]),
+            "type": "monthly",
+            "month": month
+        })
+        report = [add_user_data(serialize_doc(doc)) for doc in report]
+        return jsonify(report)
+    else:
+        #checking if join date is less than 7 or not if not subtract 7 from it
+        if datee > 7:
+            join_date = datee - 7
+        else:
+            join_date = datee
+
```

Comment:
we also need a condition that atleast 3 weekly reports has been submitted by a user before a montly report is generated. or else even new joiners will start getting this

Reply:
working on this